### PR TITLE
add client credentials grant diagram

### LINKED ISSUE #99: Interaction diagram needs updating
API Interaction sequence diagram in 2.0. APIs.md precedes recent client credentials grant and should be updated, probably with node and controller diagrams.

### PR DESCRIPTION
**Pull request by dannymeloy**
- added link to existing client credentials grant diagram
- changed the client references in the diagrams to indicate which contexts different grants might be used

Resolves #99

### CODE REVIEW

File: docs/2.0. APIs.md [hunk 1]
```diff
@@ -72,8 +72,22 @@ Server Metadata.
...
+
+### Authorization Code Grant Flow
+
+This flow can be used for any NMOS interaction.
+
 ![Authorization Flow](images/authorization_flow.png)
 
+### Client Credentials Grant Flow
+
+This flow can only be used in specific interactions. Refer to [BCP-003-02][BCP-003-02] for permitted uses of
```
Comment (prince-chrismc) [inline docs/2.0. APIs.md:86]: I only see Auth Code in BCP-003-02 https://specs.amwa.tv/bcp-003-02/branches/v1.0-dev/docs/1.0._Authorization_Practice.html#authorization-grant-constraints

Could this have a more specific link?

Comment (dannymeloy) [reply docs/2.0. APIs.md:86]: The heading "Authorization Grant Constraints" is a bit ambiguous really, as that section talks about all "Authorization Grants" not specifically the "Auth _Code_ Grant" - if anything that heading should be called "Client Credentials Grant Constraints" as it details the specific instances of when to use the CC Grant. Either way I think referencing that BCP is the correct doc

**Comment by lo-simon** [comment]
I have some comments on the diagram.

1. I think we need to add the sequence for `Request Authorization Server Metadata` before the one time Client Registration, this is also applied to the authorization_flow diagram.
2. In Client Registration, I think it may be good to specify using `private_key_jwt` for token_endpoint_auth_method and don't think Re-direct URL is required (that will require to remove `redirect_uris` from the register_client_request schema).  Using private_key_jwt, there is no Client Secert to return.
3. In Request access token, replace `Client Secret` and `Redirect URI` with  `Client Assertion` (for self signed JWT)
4. Should we also add some words regarding about client_credentials grant_type should be used for token refresh, i.e. not using the refresh_token grant for grant_type.

**Comment by dannymeloy** [comment]
I've added the metadata endpoint request as you suggested @simonlo-sony and I've also updated the diagrams to be a bit more consistent with one another.
Considering there are different methods of client authentication detailed in the spec, I've made both diagrams a bit more generic, so haven't specified any particular auth method (so steered away from specifying `private_key_jwt ` or `client_secret_post` etc) as these diagrams primary purpose is highlight the top level sequence of calls required for each grant type, so the readers can easily see the differences, without boiling down to the implementation detail.

**Comment by lo-simon** [comment]
Should the actors `User agent` and `Resource Owner` be removed too.

Need to update the register_client_request schema to remove `redirect_uris` from the required fields constrain.

**Review by andrewbonney — APPROVED** [review]

**Pull request merged**